route/table.go

```diff
@@ -290,7 +298,8 @@ func (t Table) matchingHosts(req *http.Request) (hosts []string) {
 	host := normalizeHost(req.Host, req.TLS != nil)
 	for pattern := range t {
 		normpat := normalizeHost(pattern, req.TLS != nil)
-		if glob.Glob(normpat, host) {
```

Comment:
I'm not sure why you used the glob matcher here in the first place, can a host contain wildcards ?

Reply:
Good question. I'll check.

Reply:
Yes, you can have a route for `*.example.com/foo`

Reply:
I wrote a small benchmark to verify performance. `gobwas` is fast but only when the expression is pre-compiled. It is only 1.4 ms but this sits in the hot-path. However, there is no easy way to refactor this since `Table` is just a `map[string]Routes`. Might be a good time to finally change this to a struct. I'll have a look.

```
$ go test -bench .
goos: darwin
goarch: amd64
pkg: globbench
BenchmarkGlob/ryanuber-8         	20000000	        91.4 ns/op
BenchmarkGlob/gobwas-8           	 1000000	      1385 ns/op
BenchmarkGlob/gobwas-precompile-8         	200000000	         7.29 ns/op
PASS
ok  	globbench	5.537s
```

```go
package main

import "testing"
import ryanuber "github.com/ryanuber/go-glob"
import gobwas "github.com/gobwas/glob"

var n int

func BenchmarkGlob(b *testing.B) {
	b.Run("ryanuber", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			if ryanuber.Glob("*.example.com", "foo.example.com") {
				n++
			}
		}
	})
	b.Run("gobwas", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			m := gobwas.MustCompile("*.example.com")
			if m.Match("foo.example.com") {
				n++
			}
		}
	})
	b.Run("gobwas-precompile", func(b *testing.B) {
		m := gobwas.MustCompile("*.example.com")
		for i := 0; i < b.N; i++ {
			if m.Match("foo.example.com") {
				n++
			}
		}
	})
	b.Log(n)
}
```